sympy/solvers/inequalities.py

```diff
@@ -451,6 +451,7 @@ def solve_univariate_inequality(expr, gen, relational=True, domain=S.Reals, cont
     # `solveset` makes sure this function is called only when the domain is
     # real.
     _gen = gen
+    _domain = domain
```

Comment:
at L497 & L547 there is a redefinition of domain that you are going to miss if you use the original domain. Tests for these two cases should be added to see that you are getting the right result when you use `_domain` instead of `domain` at L653.

Reply:
At L547, `continuous_domain` makes sure that the redifined domain is a subset of the original domain, so I don't think that the original domain would be missed. 
For the redefinition at L497, the test at L318 in test_inqualities.py fails in case the original domain is used. Is there a need to add a seperate test case?

Reply:
Also, if the condition at the L478 is satisfied, and the domain given is a `FiniteSet`, an exception is thrown, because `function_range` called in the next line is not yet implemented for such domains. 
https://github.com/sympy/sympy/blob/9c3a892ecfa94a4f09e71348a426a2370364e64f/sympy/solvers/inequalities.py#L478

Reply:
@smichr Should I make a seperate PR for changes in `function_range`, so that it accepts finite sets as domains, and add tests for the current PR then?